wallet-access/src/App.vue

```diff
@@ -1,21 +1,27 @@
+<template>
+  <Suspense>
+    <template #fallback>
+      Loading...
+    </template>
+    <template #default>
+      <div>
+        {{ address }}
+      </div>
+    </template>
+  </Suspense>
+</template>
+
 <script setup lang="ts">
 import { MetamaskProvider, createProvider, ChainTypes } from '@rarimo/provider'
 
-const getMetamaskWalletInfo = async () => {
-  // Connect to the Metamask wallet in the browser using Web3.js, using the MetamaskProvider interface to limit bundle size.
-  const provider = await createProvider(MetamaskProvider)
-  await provider.connect()
+// Connect to the Metamask wallet in the browser using Web3.js, using the MetamaskProvider interface to limit bundle size.
+const provider = await createProvider(MetamaskProvider)
+await provider.connect()
 
-  // Get the address of the wallet
-  console.log('Address:', provider.address)
-  console.log('chainType:', ChainTypes[provider.chainType!])
-  console.log('providerType:', provider.providerType)
-}
+// Get the address of the wallet and other info
+console.log('Address:', provider.address)
+console.log('chainType:', ChainTypes[provider.chainType!])
+console.log('providerType:', provider.providerType)
 
-getMetamaskWalletInfo()
+const address = provider.address
```

Comment:
```ts
import { ref } from 'vue'

const PROVIDER_EVENTS: Array<keyof IProvider> = [
  'onInitiated',
  'onConnect',
  'onAccountChanged',
]

const address = ref('')

 const providerEvent = provider[event] as (
        cb: (payload: ProviderEventPayload) => void,
) => void

providerEvent?.call(provider, payload => {
   address.value ||= payload?.address ?? ''
})
```

try this. PS. `const address = provider.address` is not reactive - u write value on creating component lifehook, u need to subscribe to the provider event to update value

Reply:
Looks like we need to export `ProviderEventPayload` from the SDK for this to work.

Reply:
U can use our react implementations as reference

https://github.com/rarimo/js-sdk/blob/main/packages/react-nft-checkout/src/hooks/useProvider.ts